### PR TITLE
Added USE_OSRF_REPOS build-time argument to Dockerfile (fixes #146)

### LINKED ISSUE #146: Add build-time argument to Dockerfile to use OSRF sources
Should resolve #145

### PR DESCRIPTION
**Pull request by ChrisTimperley**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -106,6 +106,13 @@ RUN pip install --upgrade \
 RUN echo "deb http://packages.osrfoundation.org/gazebo/ubuntu-stable `lsb_release -cs` main" > /etc/apt/sources.list.d/gazebo-stable.list \
  && wget http://packages.osrfoundation.org/gazebo.key -O - | apt-key add -
 
+# optionally add packages.ros.org as a source
+ARG USE_OSRF_REPOS
+RUN if [ "${USE_OSRF_REPOS}" = "True" ]; then \
+         echo "deb http://packages.ros.org/ros/ubuntu $(lsb_release -cs) main" > /etc/apt/sources.list.d/ros-latest.list \
+      && apt-key adv --keyserver hkp://ha.pool.sks-keyservers.net:80 --recv-key 421C365BD9FF1F717815A3895523BAEEB01FA116
```
Comment (gavanderhoorn) [inline Dockerfile]: the keyserver has been known to be really unreliable.

Perhaps we should consider importing the key directly from `http://packages.ros.org/ros.key` as we do with `gazebo.key`.

**Comment by ChrisTimperley** [comment]
@git-afsantos this should fix #145. I managed to build the dependencies for the PUT using the following command:
```
#!/bin/bash
pushd kobuki/9c8abeb
docker build -t cool \
  -f ../../Dockerfile \
  --build-arg UBUNTU_VERSION=precise \
  --build-arg IS_BUILD_FAILURE=True \
  --build-arg USE_APT_OLD_RELEASES=False \
  --build-arg USE_OSRF_REPOS=True \
  --build-arg ROS_DISTRO=hydro \
  --build-arg CATKIN_PKG=kobuki_gazebo_plugins \
  --build-arg REPO_FORK_URL=https://github.com/robust-rosin/kobuki_desktop \
  --build-arg REPO_BUG_COMMIT=024d79e2204a2be7ac4284bea5da2ac6ff92262a \
  --build-arg REPO_FIX_COMMIT=71f3294694ffda0e2ab48ba0842f71913204ac60 \
  . 2>&1 | tee build.log
```

**Comment by git-afsantos** [comment]
Thanks! Do you want me to try it before merging?

**Comment by git-afsantos** [comment]
I think it might be working. The build is past the previous error for me. Now it is stuck on finding the commits in the fork repo.

> fatal: Couldn't find remote ref 71f3294694ffda0e2ab48ba0842f71913204ac60

This sometimes happens, and I am not sure why. Running `docker image prune` to clear the caches solved it previously, but this time it does not seem to be working.

**Comment by ChrisTimperley** [comment]
> I think it might be working. The build is past the previous error for me. Now it is stuck on finding the commits in the fork repo.
> 
> > fatal: Couldn't find remote ref 71f3294694ffda0e2ab48ba0842f71913204ac60
> 
> This sometimes happens, and I am not sure why. Running `docker image prune` to clear the caches solved it previously, but this time it does not seem to be working.

Ah. I fixed this issue with a tweak to the Dockerfile, but that change is living on the mavros branch. I'll separate the fix and merge it into master.

**Pull request merged**